inference-engine/src/mkldnn_plugin/mkldnn_graph_optimizer.cpp

```diff
@@ -1524,64 +1524,67 @@ void MKLDNNGraphOptimizer::FuseClampAndFakeQuantize(MKLDNNGraph &graph) {
     }
 }
 
-void MKLDNNGraphOptimizer::FuseMulAddAndFakeQuantize(MKLDNNGraph &graph) {
+void MKLDNNGraphOptimizer::FusePerformedAsScaleShiftAndFakeQuantize(MKLDNNGraph &graph) {
     auto& graphNodes = graph.GetNodes();
 
-    auto isSutableScaleShiftNode = [](MKLDNNNodePtr node) {
-        return node->getType() == Eltwise && node->getChildEdges().size() == 1 && node->getAlgorithm() == EltwiseMulAdd && node->canBePerformedAsScaleShift();
+    auto getConstPort = [](const MKLDNNNodePtr node) -> int {
+        if (node->getParentEdgeAt(0)->getParent()->isConstant() && node->getParentEdgeAt(0)->getParent()->getType() == Input) {
+            return 0;
+        } else if (node->getParentEdgeAt(1)->getParent()->isConstant() && node->getParentEdgeAt(1)->getParent()->getType() == Input) {
+           return 1;
+        } else {
+            return -1;
+        }
+    };
+
+    auto isSutableScaleShiftNode = [getConstPort](MKLDNNNodePtr node) {
+        if (one_of(node->getAlgorithm(), EltwiseAdd, EltwiseMultiply, EltwiseMulAdd)) {
+            MKLDNNNode *parent = nullptr;
+            if (node->getAlgorithm() != EltwiseMulAdd) {
+                const auto constPort = getConstPort(node);
+                if (constPort == -1) {
+                    return false;
+                }
+                parent = node->getParentEdgeAt(1 - constPort)->getParent().get();
+            }
+            return node->getType() == Eltwise && node->getChildEdges().size() == 1 && node->canBePerformedAsScaleShift(parent);
```

Comment:
Why canBePerformedAsScaleShift is called for parent of Eltwise operation?

Reply:
```canBePerformedAsScaleShift``` called for ScaleShift candidate, ```parent``` used inside ```canBePerformedAsScaleShift``` to separate data input and const nodes